docs/testplan/MACsec-test-plan.md

```diff
@@ -0,0 +1,445 @@
+
+#### Verify SNMP request/response works across interface with macsec configuration
+
+- Configure the macsec profile on interface and check if the snmp walk succeeds from the peer VM.
+
+### Testcase : Deployment usecases and fault handling scenario's
+
+  This testcase covers the various fault scenario's and the expected behavior.
+
+#### Link flap on an interface with macsec configured
+
+- MKA session can be recovered from the link flap if the port comes back up in < 6 secs (MKA lietime)
+- If the port is down for more than 6 sec, MKA session will create a new session.
+
+#### Link flap on a portchannel member which has macsec configured
+
+- When the member interface flaps and it is the only portchannel member
+   - Expect the Portchannel to go down and come up depending on whether the member port comes back in 6 secs (MKA lietime)
+- When one member interface flaps, but the Portchannel has more member ports which are macsec enabled.
+   - Expect Portchannel to remain up.
+
+#### MACsec session cannot be established under wrong MKA configuration
+
+- If the CAK is mis-matched between DUT and peer, the MKA session cannot be established.
+   - Control plane protocols eg: BGP session will not be established.
+
+#### Config reload done on DUT with macsec configuration
+
+- The macsec sessions will be reconfigured, MKA session created again with a new SAK key.
+- The control protocol sessions like LACP, LLDP, BGP get established again over the macsec configured interfaces.
+
+#### Everflow, port mirroring on macsec enabled interfaces
+
+- This test is to verify 2 cases
+   - configure mirroring on interfaces where macsec is configured.
+   - configure macsec on the outgoing interface ( in case of Everflow ).
+     - TODO add expected behavior
+
+### Testcase : Macsec scenario's for multi-asic, multi-dut
+
+- Verify that macsec dockers are coming up in different namespaces.
+- Verify macsec packet flow where the Ingress and Egress ports are on different Linecards.
+   - TODO add expected behavior
+
+### Testcase : Scale tests
+
+#### Large number of interfaces having macsec enabled on the DUT/linecard
+
+- Use the "configure feature macsec enabled" command to turn ON macsec, with all ports having macsec profile attached.
+   - Check the CPU, ASIC behavior when there are multiple wpa_supplicant processes being spawned.
+   - Expect the macsec sessions all come up. Measure the time taken.
+- When all the interfaces flap together, how much time it takes for Portchannels/BGP sessions to be up
+    -Expect the macsec sessions all come up. Measure the time taken.
+
+#### Simultaneous rekeying in all macsec sessions
+
+- Enable macsec on multiple interfaces so that they have same rekey period.
+- Check the sessions are coming back up and there is no traffic loss during h/w progamming.
+
+#### Macsec enabled on all interfaces and the DUT is rebooted
+
+- Check the macsec docker comes up and macsec sessions are established.
```

Comment:
Any test coverage for counters and stats?

Reply:
Yes, I needed it. I added a section for testing counters and stats and will fill it later.